cli/cmd/util.go

```diff
@@ -51,58 +51,67 @@ var (
-	// skip verification if specified
-	if insecure {
-		fmt.Println("Warning: skipping quote verification")
-		return era.InsecureGetCertificate(host)
-	}
-
-	// get certificate using provided config
-	if configFilename != "" {
-		pemBlock, tcbStatus, err := era.GetCertificate(host, configFilename)
-		if errors.Is(err, attestation.ErrTCBLevelInvalid) && util.StringSliceContains(acceptedTCBStatuses, tcbStatus.String()) {
-			fmt.Println("Warning: TCB level invalid, but accepted by configuration")
-			return pemBlock, nil
-		}
-		return pemBlock, err
-	}
-
-	// get latest config from github if none specified
+func fetchLatestCoordinatorConfiguration() error {
 	coordinatorVersion, err := getCoordinatorVersion()
 	eraURL := fmt.Sprintf("https://github.com/edgelesssys/marblerun/releases/download/%s/coordinator-era.json", coordinatorVersion)
 	if err != nil {
 		// if errors were caused by an empty kube config file or by being unable to connect to a cluster we assume the Coordinator is running as a standlone
 		// and we default to the latest era-config file
 		var dnsError *net.DNSError
 		if !clientcmd.IsEmptyConfig(err) && !errors.As(err, &dnsError) && !os.IsNotExist(err) {
-			return nil, err
+			return err
 		}
 		eraURL = "https://github.com/edgelesssys/marblerun/releases/latest/download/coordinator-era.json"
 	}
 
 	fmt.Printf("No era config file specified, getting config from %s\n", eraURL)
 	resp, err := http.Get(eraURL)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("downloading era config failed with error %d: %s", resp.StatusCode, http.StatusText(resp.StatusCode))
+		return fmt.Errorf("downloading era config failed with error %d: %s", resp.StatusCode, http.StatusText(resp.StatusCode))
 	}
 	out, err := os.Create(eraDefaultConfig)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer out.Close()
 	_, err = io.Copy(out, resp.Body)
 	if err != nil {
-		return nil, err
+		return err
 	}
+
 	fmt.Println("Got latest config")
+	return nil
+}
+
+// verify the connection to the MarbleRun Coordinator.
+func verifyCoordinator(host string, configFilename string, insecure bool, acceptedTCBStatuses []string) ([]*pem.Block, error) {
+	// skip verification if specified
+	if insecure {
+		fmt.Println("Warning: skipping quote verification")
+		return era.InsecureGetCertificate(host)
+	}
 
-	pemBlock, _, err := era.GetCertificate(host, eraDefaultConfig)
+	var configFile string
+	// get certificate using provided config
+	if configFilename != "" {
+		configFile = configFilename
+	} else {
+		// get latest config from github if none specified
+		if err := fetchLatestCoordinatorConfiguration(); err != nil {
+			return nil, err
+		}
+		configFile = eraDefaultConfig
+	}
+
+	pemBlock, tcbStatus, err := era.GetCertificate(host, configFile)
```

Comment:
```suggestion
	if configFilename == "" {
		// get latest config from github if none specified
		if err := fetchLatestCoordinatorConfiguration(); err != nil {
			return nil, err
		}
		configFilename = eraDefaultConfig
	}

	pemBlock, tcbStatus, err := era.GetCertificate(host, configFilename)
```
nit: IMO this is more readable without introducing a new variable. But if you disagree, we could also keep it as is.

Reply:
Thanks, you are right! :100: